content/tutorials/portenta-h7/por-openmv-gs/content.md

```diff
@@ -0,0 +1,211 @@
+
+***Installing the OpenMV firmware will overwrite any existing sketches in the internal flash of Portenta. As a result the M7 port won't be exposed in the Arduino IDE anymore. To re-flash the M7 with an Arduino firmware you need to put the board into bootloader mode. To do so double press the reset button on the Portenta H7 board. The built-in green LED will start fading in and out. In bootloader mode you will see the Portenta M7 port again in the Arduino IDE.***
+
+The Portenta will start flashing its blue LED when it's ready to be connected. After confirming the completion dialog the Portenta should already be connected to the OpenMV IDE, otherwise click the "connect" button once again.
+
+![When the Portenta is successfully connected to the OpenMV IDE a green play button appears in the lower left](assets/por_openmv_board_connected.png)
+
+### 3. Wiring the LED to the Portenta
+
+To be able to change the LED's intensity with PWM we need to first wire the Portenta to the LED. If you want to use other pins than the ones shown here, take a look at the pinout diagram.
+
+![Portenta pinout](assets/Pinout-PortentaH7_latest.png)
+
+Here you can also see what functionality each pin has. In this tutorial wire the pin PC6 via a resistor to the positive leg of the LED, and wire the boards GND to the negative leg of the LED. As shown in the illustration.
+
+[Illustration of a LEDs positive leg on a breadboard wired to the PC6 pin on the Portenta, and the negative end going to the GND pin on the Portenta]()
+
+### 4. Preparing the MicroPython Sketch
+
+Create a new script by clicking the "New File" button in the toolbar on the left side. Import the required modules:
+
+```py
+import pyb # Import module for board related functions
+import time # Import module for board related functions
+from pyb import Pin, Timer # Import module for board related functions
+```
+
+A module in Python is a confined bundle of functionality. By importing it into the script it gets made available.
+
+### 5. Preparing the Pin and PWM
+
+In order to control a LED connected to one of our pins, we first need to configure the pins behavior. After, we can set up the PWM.
+
+```py
+pin1 = Pin("PC6", Pin.OUT_PP, Pin.PULL_NONE)
+timer1 = Timer(3, freq=1000) # Frequency in Hz
+channel1 = timer1.channel(1, Timer.PWM, pin=pin1, pulse_width=0)
+
+# maximum and minimum pulse-width, corresponds to maximum and minimum brightness
+maxWidth = 5000
+minWidth = 0
+
+# How much to change the pulse-width by each step
+step = 500
+# Set starting value
+curWidth = minWidth
+```
+
+In the `pin1` variable we define what pin on the board we intend to use. We also define if it should be an input or output pin with `Pin.OUT_PP`. Lastly, we can set the behavior of the pull up resistor on the pin, for now we set it as `Pin.PULL_NONE`.
```

Comment:
Its a good practice to give the explanation first and then the code. here you are writing the code first followed by the explanation. As a user I wouldnt understand why am i including certain variables